crates/core/component/dex/src/component/rpc.rs

```diff
@@ -541,7 +543,13 @@ impl SimulationService for Server {
             .await
             .map_err(|e| tonic::Status::internal(format!("error simulating trade: {:#}", e)))?;
 
+        let unfilled = Value {
+            amount: input.amount - swap_execution.input.amount,
```

Comment:
This all looks good but can we make this use `checked_sub` and log the error, just in case we turn up bugs in `SwapExecution`.

Reply:
Ah, good call. Will do.